stubs/boltons/boltons/cacheutils.pyi

```diff
@@ -0,0 +1,107 @@
+from _typeshed import Incomplete
+from collections.abc import Callable, Generator
+from typing import Any, NoReturn, TypeVar
+
+_KT = TypeVar("_KT")
+_VT = TypeVar("_VT")
+
+xrange = range
+basestring: Any
+PREV: int
+NEXT: int
+KEY: int
+VALUE: int
+DEFAULT_MAX_SIZE: int
+
+class LRI(dict[_KT, _VT]):
+    hit_count: int
+    max_size: int
+    on_miss: Callable[[_KT], _VT]
+    def __init__(self, max_size: int = ..., values: Incomplete | None = ..., on_miss: Incomplete | None = ...) -> NoReturn: ...
+    def __setitem__(self, key: _KT, value: _VT) -> NoReturn: ...
```

Comment:
```suggestion
    def __setitem__(self, key: _KT, value: _VT) -> None: ...
```
`NoReturn` should be used if the function never returns (e.g. if it throws an exception), not if it simply returns None. This probably affects a bunch of other places; please fix them too.